OuterWildsRandomSpeedrun/SpawnPointPool.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace OuterWildsRandomSpeedrun
+{
+    public struct SpawnPointConfig
+    {
+        public string internalId;
+        public string displayName;
+    }
+
+    public class SpawnPointPool
+    {
+        public List<SpawnPointConfig> SpawnPointConfigs
+        {
+            get => _spawnPointConfigs;
+        }
+
+        private List<SpawnPointConfig> _spawnPointConfigs;
+
+        private SpawnPointPool(List<SpawnPointConfig> configs)
+        {
+            this._spawnPointConfigs = configs;
+        }
+
+        public static SpawnPointPool FromTsv(string pathToTsv)
+        {
+            var configs = ParseTsv(pathToTsv).Select(line => BuildSpawnPointConfig(line)).ToList();
+            return new SpawnPointPool(configs);
+        }
+
+        private static SpawnPointConfig BuildSpawnPointConfig(string[] line) =>
+            new SpawnPointConfig { internalId = line[0], displayName = line[1] };
+
+        private readonly static char[] LINE_SEPARATORS = new char[] { '\r', '\n' };
+        private readonly static char FIELD_SEPARATOR = '\t';
+        private static string[][] ParseTsv(string pathToTsv)
+        {
+            using (var reader = new StreamReader(pathToTsv))
+            {
+                var contents = reader.ReadToEnd();
+                var lines = contents.Split(LINE_SEPARATORS, StringSplitOptions.RemoveEmptyEntries);
+                return lines
+                    .Skip(1) // Ignore the header row
+                    .Select(line => line.Split(FIELD_SEPARATOR))
+                    .ToArray();
```

Comment:
It's an `IEnumerable`. In general I'm wary of exposing those in APIs since the underlying object might be single-use, which could cause problems if callers don't expect that behavior. Given that this list is ~150 entries and will be read exactly once, I'm not too concerned about perf.

Reply:
That said, this is a private method that's called exactly once, so I can make the change if you feel strongly about it.

Reply:
nope, I don't feel strongly. Generally though, I expect code within a class to be aware of the inner workings elsewhere in the class, so I wouldn't be concerned about the single-use thing.